cmake/CompilerSetup.cmake

```diff
@@ -32,6 +32,9 @@ if(CLANG OR GNU)
     common_flags
     -Wall -Wextra -Werror
 
+    # Detect possible misuse of comma operator.
```

Comment:
Nit: rather than commenting on each of these individually you might consider just a bucket for "match Apple recommended project settings" or something like that.

Reply:
will do. I've got one further commit for this PR anyways: This option is clang only and won't work with gcc, so I'll move it down to the clang section below.

I'd upload it now, but I'm at the repair shop and their wifi seems to block ssh. :/ (Should be here within a few hours though.)